### DISCUSSION TITLE
Pulsar SQL support for Apache Pulsar 3.2+

### DISCUSSION THREAD

**Discussion by Sandhya-Sasikumar** [apache/pulsar#23435] [Q&A] [answered]
Hi, When i look at the pulsar documentation I dont see pulsar sql from pulsar 3.2+ version onwards. Any idea wheather pulsar sql is  available as seperate package? I couldn't locate. Please advise if any one have any idea on that 

https://pulsar.apache.org/docs/3.2.x/   -- Pulsar SQL is missing here

**Comment by lhotari** [comment, marked as answer]
Pulsar SQL has been removed from the apache/pulsar repository. Some history:

* In April 2020, it was decided to remove Pulsar SQL from apache/pulsar repository:
  * https://lists.apache.org/thread/qr49rp91w8pp8ljz1jtxn92ktnznh9tb
  * https://github.com/apache/pulsar/wiki/PIP-62%3A-Move-connectors%2C-adapters-and-Pulsar-Presto-to-separate-repositories
* In April 2021, there was a decision to contribute Pulsar SQL as a Trino plugin:
  * https://lists.apache.org/thread/f9n6sc2mrboq5sxhjbr7gvdl8vqp9fpk
* This was the status in December 2021:
  * https://lists.apache.org/thread/gqs9tq3p6tr38b7j6gjbfzfg7z1vkvsd
  * The Trino PR was https://github.com/trinodb/trino/pull/8020
  * The work on the plugin stalled and it was never taken to completion.
* In December 2023, Pulsar SQL was finally removed from the apache/pulsar repository:
  * https://lists.apache.org/thread/4f1cco12cycq36m7vtyjs2j5q5975666
* Today, October 18, 2024, Pulsar SQL has been forked from apache/pulsar branch-3.0 to https://github.com/apache/pulsar-sql

There are 3 options to continue to use Pulsar SQL:
- **Use Pulsar 3.0.x LTS which includes Pulsar SQL.** [Pulsar 3.0.x LTS is supported and maintained until May 2025](https://pulsar.apache.org/contribute/release-policy/#supported-versions).
- Contribute to the forked https://github.com/apache/pulsar-sql repository so that it could be maintained beyond May 2025 by the Pulsar community. Contributors are welcome to join this effort! The first task is to setup the maven build for the project. It's missing.
- Contribute to the trino-pulsar plugin
  - Recently @eaba has resumed the original trino-pulsar PR 8020 in https://github.com/trinodb/trino/pull/23566. More contributors are welcome to join this effort.

Contributions are welcome! We can coordinate the efforts on Pulsar Slack's #dev channel and create a separate #dev-trino channel for coordinating the contributions. @Sandhya-Sasikumar Would you be interested in contributing?
